src/sos/objectstorage.go

```diff
@@ -85,21 +80,99 @@ func (o *ObjectStorage) fetchManagedBuckets(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("cannot list buckets: %w", err)
 		}
-		bucketDetailsByCluster := matchOrgAndNamespaceByBucket(ctx, &buckets, k8sCluster.Name)
+		bucketDetailsByCluster := addOrgAndNamespaceToBucket(ctx, &buckets, k8sCluster.Name)
 
 		// add BucketDetail from each cluster to one single slice
 		o.bucketDetails = append(o.bucketDetails, *bucketDetailsByCluster...)
 	}
 	return nil
 }
 
-func matchOrgAndNamespaceByBucket(ctx context.Context, buckets *exoscalev1.BucketList, clusterName string) *[]BucketDetail {
+func (o *ObjectStorage) saveToDatabase(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Creating a database connection")
+
+	log.V(1).Info("Opening database connection")
+	err := o.database.OpenConnection()
+	if err != nil {
+		return err
+	}
+	defer o.database.CloseConnection()
+
+	log.V(1).Info("Ensuring initial database configuration")
+	err = o.database.EnsureInitConfiguration(ctx)
```

Comment:
I don't like the fact that the DB is re-initialized for every entry that gets stored to the DB. It is an unnecessary overhead and also conceptually just the wrong place to do this.

In practice it most likely won't matter much though.

Reply:
Actually It runs once from my perspective. Can you explain what is wrong? maybe I'm missing something.

Reply:
@ccremer do you see any issues here?

Reply:
The current implementation doesn't look like it's re-initialized for every entry. It's outside of the for-loop that appears in the later lines.